### ISSUE TITLE
Mon failover takes an extra cycle to remove the failed mon

### ISSUE THREAD

**Issue by travisn** [rook/rook#2378]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
When a mon is failed over, a new mon is added to quorum ok, but the failed mon takes a couple minutes to be removed from the monmap. Messages in the operator log show that the operator is waiting incorrectly for the wrong number of mons to start. After that wait times out, the health check at the next interval will properly remove the mon.

```
2018-12-13 09:00:38.318516 W | op-mon: mon b not found in quorum, still in mon out timeout
2018-12-13 09:00:53.654322 W | op-mon: mon b not found in quorum, still in mon out timeout
2018-12-13 09:01:09.002104 W | op-mon: mon b NOT found in quorum and timeout exceeded, mon will be failed over
2018-12-13 09:01:09.002240 I | op-mon: Failing over monitor b
2018-12-13 09:01:09.002268 I | op-mon: starting new mon: &{ResourceName:rook-ceph-mon-d DaemonName:d PublicIP: Port:6790}
2018-12-13 09:01:09.032005 I | op-mon: mon d running at 10.102.210.232:6790
2018-12-13 09:01:09.049357 I | op-mon: Found 0 running nodes without mons
2018-12-13 09:01:09.051131 I | op-mon: All nodes are running mons. Adding all 1 nodes to the availability.
2018-12-13 09:01:09.067013 I | op-mon: mons created: 1
2018-12-13 09:01:09.067220 I | op-mon: waiting for mon quorum with [d]
2018-12-13 09:01:09.085146 I | op-mon: 2/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:14.092464 I | op-mon: 2/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:19.103375 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:24.111732 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:29.121748 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:34.126907 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:39.146381 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:44.152441 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:49.159388 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:54.174000 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:01:59.183581 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:04.189701 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:09.198285 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:14.205445 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:19.215370 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:24.223625 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:29.234446 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:34.242159 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:39.250987 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:44.259682 I | op-mon: 3/1 mon pods are running. waiting for pods to start
2018-12-13 09:02:44.260065 E | op-mon: failed to failover mon b. failed to start new mon d. failed to wait for mon quorum. exceeded max retry count waiting for monitors to reach quorum
2018-12-13 09:02:59.566501 W | op-mon: mon b NOT found in quorum and timeout exceeded, mon will be failed over
2018-12-13 09:02:59.566542 I | op-mon: ensuring removal of unhealthy monitor b
2018-12-13 09:02:59.572571 I | op-mon: dead mon rook-ceph-mon-b was already gone
2018-12-13 09:02:59.572995 I | exec: Running command: ceph mon remove b --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/615018395
2018-12-13 09:02:59.881910 I | exec: removing mon.b at 10.108.158.149:6790/0, there will be 3 monitors
2018-12-13 09:02:59.882682 I | op-mon: removed monitor b
2018-12-13 09:02:59.937899 I | op-mon: saved mon endpoints to config map map[data:a=10.105.205.66:6790,c=10.96.157.62:6790,d=10.102.210.232:6790 maxMonId:2 mapping:{"node":{"a":{"Name":"minikube","Hostname":"minikube","Address":"10.0.2.15"},"c":{"Name":"minikube","Hostname":"minikube","Address":"10.0.2.15"},"d":{"Name":"minikube","Hostname":"minikube","Address":"10.0.2.15"}},"port":{}}]
2018-12-13 09:02:59.938649 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2018-12-13 09:02:59.939048 I | cephconfig: copying config to /etc/ceph/ceph.conf
2018-12-13 09:02:59.939486 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2018-12-13 09:02:59.944441 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2018-12-13 09:02:59.946493 I | cephconfig: copying config to /etc/ceph/ceph.conf
2018-12-13 09:02:59.946899 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
```

**Expected behavior:**
The failed mon should be removed from the monmap as part of the initial operation instead of at the next health check.

**How to reproduce it (minimal and precise):**

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)):

**Comment by immesys** [comment]
Hmm I think I am observing this too. 

I had a healthy cluster, then I cordoned a node running `mon a`. The rook operator noticed and started a new mon, but then left `mon a` running. So now the ceph cluster has 4 mons in quorum. I would have expected it to remove the "failed" mon. I haven't left it for very long (only 15 mins) so perhaps it will remove it after a few more minutes?

**Comment by immesys** [comment]
Even restarting the operator is not helping here.

**Comment by immesys** [comment]
I am on 0.9.2

**Comment by travisn** [comment]
Fixed by #2631
